actor/future.go

```diff
@@ -46,27 +46,37 @@ func (f *Future) PID() *PID {
 
 // PipeTo forwards the result or error of the future to the specified pids
 func (f *Future) PipeTo(pids ...*PID) {
+	f.cond.L.Lock()
 	f.pipes = append(f.pipes, pids...)
-}
-
-func (f *Future) sendToPipes() {
-	if f.pipes == nil {
-		return
+	//for an already completed future, force push the result to targets
+	if f.done {
+		f.tell()
 	}
+	f.cond.L.Unlock()
+}
 
+func (f *Future) tell() {
```

Comment:
this is a new internal method for Futures to send to all the recipients.
This is called both when the Future completes, and if anyone calls PipeTo after the Future is done.